### ISSUE TITLE
FR: Documentation for out of core operations using the .jay format

### ISSUE THREAD

**Issue by ChristopherEeles** [h2oai/datatable#3147]
Hello datatable team,

Great work so far! As a long time R user I appreciate the familiarity data.table when working in Python.

You recently added your memory mapped binary format, .jay.

Do you have plans to document the out of core API any time soon? Things like which operations are supported out of core and how common operations would be achieved.

As a particular question, I wonder if it is possible to rbind many .jay datatables with a memory limit?

Best,
Christopher Eeles
Software Developer @ BHKLab
PMCC @ UHN

**Comment by oleksiyskononenko** [comment]
`rbind()` was designed to rbind frames, not jay files. Also, it doesn't take into account how a particular frame was created — it just allocates a memory buffer for each column and then copies all the frames' content into those buffers. 

As for your first question: any operation that produces virtual columns as a result, will not increase a memory footprint. This is currently not documented as virtual columns are transparent for users, and also some operations are still to be improved to avoid materialization and data copying. Some features we did documented, like the [memory limit](https://datatable.readthedocs.io/en/latest/api/dt/fread.html#memory_limit) options or the jay files handling.

**Comment by ChristopherEeles** [comment]
Hi @oleksiyskononenko,

Sorry for writing imprecisely, I do not wish to append .jay files. 

I was simply wondering how `dt.rbind` responds to concatenating tables loaded FROM .jay files, and if it is memory limit aware.

For example, I have noticed that even without a memory limit, loading a .jay file into a `dt.Frame` with `fread` does not load the entire table into memory. Indeed it seems that most operations on a `dt.Frame` loaded in this manner have a very low memory footprint relative to the size of the table of disk.

This leads me to the question: which operations will materialize the table and which won't? 

Currently (at least in the version of datatable 1.0.0 I compiled locally on Ubuntu 20.04) it seems like `dt.rbind` DOES materialize the tables, at least when loading a table which has been partitioned into multiple .jay files with `df = dt.rbind(*dt.iread(path_to_tables))`.

Given that loading partitioned tables is a pretty common data munging activity, I wondered if there are plans for `dt.rbind` to become memory limit aware and if so, how that would work. 

For example, I could imagine calling `dt.iread` with a memory limit argument and the subsequent `dt.rbind` call could do three things: (1) ignore the memory limit and materialize all tables (I believe this is the current behaviour), (2) concatenate the tables respecting the memory limit per table (num_tables * memory_limit = total_footprint), (3) concatenate the tables, respecting the memory limit specified to `dt.iread`.

I do not think the behaviour in such situations is apparent from the current documentation, and thus am asking:

1. If adding documentation and examples of out of core operations is on the current road map, and
2. What kind of timeline we could expect to see such documentation, if it is planned.

I am already using `datatable` in a production ETL pipeline and would like to get a deeper understanding of out of core operations in general.

Thanks for your response.

Best,
Christopher Eeles
Software Developer @ BHKLab
PMCC @ UHN
